micromixin-transformer/src/main/java/org/stianloader/micromixin/transform/internal/annotation/MixinShadowAnnotation.java

```diff
@@ -107,14 +124,24 @@ private void apply(@NotNull MixinFieldStub source, @NotNull ClassNode target, @N
 
     @Override
     public void apply(@NotNull ClassNode to, @NotNull HandlerContextHelper hctx,
-            @NotNull MixinStub sourceStub, @NotNull T source, @NotNull SimpleRemapper remapper,
-            @NotNull StringBuilder sharedBuilder) {
-        // NOP
+                      @NotNull MixinStub sourceStub, @NotNull T source, @NotNull SimpleRemapper remapper,
+                      @NotNull StringBuilder sharedBuilder) {
+        if (!this.isMutable) return;
+
+        if (source instanceof MixinFieldStub) {
+            for (FieldNode fn : to.fields) {
+                if (fn.name.equals(source.getName()) && fn.desc.equals(source.getDesc())) {
```

Comment:
This does not properly account for prefixes and aliases. Use the `remapper` instead on the source.getName/source.getDesc

Reply:
Am I supposed to access the fieldRenames field and use the methods from there? If yes, does that method have to take in the remapped or the original field descriptor?

Reply:
oh, and it's always the source descriptor